server/src/main/java/de/uftos/repositories/solver/timefold/constraints/ConstraintDefinitionFactory.java

```diff
@@ -0,0 +1,1073 @@
+  }
+
+  private static Function<List<ResourceTimefoldInstance>, Number> convertSize(
+      AbstractSyntaxTreeDto ast, LinkedHashMap<String, ResourceType> params) {
+    if (ast.getToken() != UcdlToken.SIZE) {
+      throw new IllegalStateException();
+    }
+    OperatorDto size = (OperatorDto) ast;
+    if (size.parameters().size() != 1) {
+      throw new IllegalStateException();
+    }
+    Function<List<ResourceTimefoldInstance>, Set<ResourceTimefoldInstance>> set =
+        convertSet(size.parameters().getFirst(), params);
+    return (parameters) -> new Number(set.apply(parameters).size());
+  }
+
+  private static Function<List<ResourceTimefoldInstance>, Set<ResourceTimefoldInstance>> convertSet(
+      AbstractSyntaxTreeDto ast, LinkedHashMap<String, ResourceType> params) {
+    return switch (ast.getToken()) {
+      case RESOURCE_SET -> convertResourceSet(ast, params);
+      case NUMBER_SET -> convertNumberSet(ast, params);
+      default -> throw new IllegalStateException();
+    };
+  }
+
+  private static Function<List<ResourceTimefoldInstance>, Set<ResourceTimefoldInstance>>
+      convertResourceSet(AbstractSyntaxTreeDto ast, LinkedHashMap<String, ResourceType> params) {
+    if (ast.getToken() != UcdlToken.RESOURCE_SET) {
+      throw new IllegalStateException();
+    }
+    SetDto set = (SetDto) ast;
+
+    Function<List<ResourceTimefoldInstance>, ResourceTimefoldInstance> name;
+
+    if (set.setName().getToken() == UcdlToken.VALUE_REFERENCE) {
+      name = convertValueReference(set.setName(), params);
+    } else {
+      name = convertElement(set.setName(), params);
+    }
+
+    List<Function<
+        List<ResourceTimefoldInstance>,
+        Function<Set<ResourceTimefoldInstance>, Set<ResourceTimefoldInstance>>
+        >> modifiers = new ArrayList<>();
+
+    for (AbstractSyntaxTreeDto modifier : set.modifiers()) {
+      if (modifier.getToken() == UcdlToken.ATTRIBUTE) {
+        modifiers.add(convertAttribute(modifier, params));
+      } else {
+        modifiers.add(convertFilter(modifier, params));
+      }
+    }
+
+    return (parameters) -> {
+      Set<ResourceTimefoldInstance> setReference = new HashSet<>();
+      setReference.add(name.apply(parameters));
+
+      for (Function<
+            List<ResourceTimefoldInstance>,
+            Function<Set<ResourceTimefoldInstance>, Set<ResourceTimefoldInstance>>
+            > function : modifiers) {
+        setReference = function.apply(parameters).apply(setReference);
+      }
+      return setReference;
+    };
+  }
+
+  private static Function<List<ResourceTimefoldInstance>, Set<ResourceTimefoldInstance>>
+      convertNumberSet(AbstractSyntaxTreeDto ast, LinkedHashMap<String, ResourceType> params) {
+    if (ast.getToken() != UcdlToken.NUMBER_SET) {
+      throw new IllegalStateException();
+    }
+    ValueDto<Integer[]> set = (ValueDto<Integer[]>) ast;
+
+    Set<ResourceTimefoldInstance> resources = new HashSet<>();
+
+    for (Integer i : set.value()) {
```

Comment:
```suggestion
    for (int i : set.value()) {
```
You shouldn't use wrapper types if avoidable

Reply:
I guess that should work too.
I'll fix this.